lib/Doctrine/Inflector/Pluralizer.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Inflector;
+
+use Doctrine\Inflector\Rules\Plural;
+use Doctrine\Inflector\Rules\Uninflected;
+use function array_keys;
+use function array_merge;
+use function implode;
+use function preg_match;
+use function preg_replace;
+use function strtolower;
+use function substr;
+
+class Pluralizer extends InflectorService
+{
+    /** @var Uninflected */
+    private $uninflected;
+
+    public function __construct(Uninflected $uninflected)
+    {
+        $this->uninflected = $uninflected;
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRules() : array
+    {
+        return Plural::RULES;
+    }
+
+    /**
+     * Returns a word in plural form.
+     *
+     * @param string $word The word in singular form.
+     *
+     * @return string The word in plural form.
+     */
+    public function inflect(string $word) : string
+    {
+        if (isset($this->cache['pluralize'][$word])) {
+            return $this->cache['pluralize'][$word];
+        }
+
+        if (! isset($this->rules['merged']['irregular'])) {
+            $this->rules['merged']['irregular'] = $this->rules['irregular'];
+        }
+
+        if (! isset($this->rules['merged']['uninflected'])) {
+            $this->rules['merged']['uninflected'] = array_merge(
+                $this->rules['uninflected'],
+                $this->uninflected->getUninflected()
+            );
+        }
+
+        if (! isset($this->rules['cacheUninflected']) || ! isset($this->rules['cacheIrregular'])) {
+            $this->rules['cacheUninflected'] = '(?:' . implode('|', $this->rules['merged']['uninflected']) . ')';
+            $this->rules['cacheIrregular']   = '(?:' . implode('|', array_keys($this->rules['merged']['irregular'])) . ')';
+        }
+
+        if (preg_match('/(.*)\\b(' . $this->rules['cacheIrregular'] . ')$/i', $word, $regs)) {
+            $this->cache['pluralize'][$word] = $regs[1] . $word[0] . substr($this->rules['merged']['irregular'][strtolower($regs[2])], 1);
+
+            return $this->cache['pluralize'][$word];
+        }
+
+        if (preg_match('/^(' . $this->rules['cacheUninflected'] . ')$/i', $word, $regs)) {
+            $this->cache['pluralize'][$word] = $word;
+
+            return $word;
+        }
+
+        foreach ($this->rules['rules'] as $rule => $replacement) {
+            if (preg_match($rule, $word)) {
+                $this->cache['pluralize'][$word] = preg_replace($rule, $replacement, $word);
+
+                return $this->cache['pluralize'][$word];
+            }
+        }
```

Comment:
As Scrutinizer points out, this will implicitly return `null` which is incompatible with non-null return type.